File: exercises/small_problems_101/easy_4/short_long_short.rb
def short_long_short(string1, string2)
  long = string1.length > string2.length ? string1 : string2
  short = string1.length < string2.length ? string1 : string2
  short + long + short
end

puts short_long_short('abc', 'defgh') == "abcdefghabc"
puts short_long_short('abcde', 'fgh') == "fghabcdefgh"
puts short_long_short('', 'xyz') == "xyz"

File: frontend/javascript/exercises/easy_1/short_long_short.js
function getLongerStr(str1, str2) {
  return str1.length > str2.length ? str1 : str2;
}

function getShorterStr(str1, str2) {
  return str1.length < str2.length ? str1 : str2;
}

function shortLongShort(str1, str2) {
  var shorterStr = getShorterStr(str1, str2);
  var longerStr = getLongerStr(str1, str2);
  return shorterStr + longerStr + shorterStr;
}

console.log(shortLongShort('abc', 'defgh') === 'abcdefghabc');
console.log(shortLongShort('abcde', 'fgh') === 'fghabcdefgh');
console.log(shortLongShort('', 'xyz') === 'xyz');
